darshan-util/pydarshan/darshan/tests/test_report.py

```diff
@@ -441,7 +441,7 @@ def test_runtime_dxt_heatmap_similarity():
 
     # DXT heatmap:
     agg_df = heatmap_handling.get_aggregate_data(report=report, ops=["write"])
-    dxt_heatmap_df = heatmap_handling.get_heatmap_df(agg_df=agg_df, xbins=32, nprocs=32)
+    dxt_heatmap_df = heatmap_handling.get_heatmap_df(agg_df=agg_df, xbins=34, nprocs=32, tmax=3.4)
```

Comment:
Why are we feeding in a magic number like `3.4` in here? It should always be possible to conveniently calculate values like this programmatically--in this case, if the default time precision isn't good enough, just round it up to prevent truncating the data unnecessarily, and apply the same policy globally (pad a 1 second resolution timer by + 1 second). This isn't just some weird test case--in theory, there could be an infinite number of logs that would have truncated data if we don't round up.

Honestly, I'm not a huge fan of discarding the time resolution DXT gives us instead of devising an approach to use the best resolution we have available with the heatmaps that a given report contains, i.e., the max time across all the different heatmap dataframes for example. Maybe it could later be relaxed when the runtime resolution improves.

Reply:
Yeah I see your point. I'll update this to use the report metadata when I pad `tmax`.